lang/rust/rust.py

```diff
@@ -0,0 +1,210 @@
+    'you thirty two': 'u32',
+    'eye thirty two': 'i32',
+    'you sixty four': 'u64',
+    'eye sixty four': 'i64',
+    'you size': 'usize',
+    'eye size': 'isize',
+    'boolean': 'bool',
+    'string': 'str',
+}
+
+@ctx.action_class('user')
+class UserActions:
+    def code_operator_lambda():
+        actions.insert('|| ')
+
+    def code_operator_subscript():
+        actions.insert('[]')
+        actions.edit.left()
+
+    def code_operator_increment():
+        actions.insert(' += 1')
+
+    def code_operator_indirection():
+        actions.auto_insert('*')
+
+    def code_operator_address_of():
+        actions.auto_insert('&')
+
+    def code_operator_assignment():
+        actions.auto_insert(' = ')
+
+    def code_operator_subtraction():
+        actions.auto_insert(' - ')
+
+    def code_operator_subtraction_assignment():
+        actions.auto_insert(' -= ')
+
+    def code_operator_addition():
+        actions.auto_insert(' + ')
+
+    def code_operator_addition_assignment():
+        actions.auto_insert(' += ')
+
+    def code_operator_multiplication():
+        actions.auto_insert(' * ')
+
+    def code_operator_multiplication_assignment():
+        actions.auto_insert(' *= ')
+
+    def code_operator_exponent():
+        actions.auto_insert('.pow()');
+        actions.edit.left();
+
+    def code_operator_division():
+        actions.auto_insert(' / ')
+
+    def code_operator_division_assignment():
+        actions.auto_insert(' /= ')
+
+    def code_operator_modulo():
+        actions.auto_insert(' % ')
+
+    def code_operator_modulo_assignment():
+        actions.auto_insert(' %= ')
+
+    def code_operator_equal():
+        actions.auto_insert(' == ')
+
+    def code_operator_not_equal():
+        actions.auto_insert(' != ')
+
+    def code_operator_greater_than():
+        actions.auto_insert(' > ')
+
+    def code_operator_greater_than_or_equal_to():
+        actions.auto_insert(' >= ')
+
+    def code_operator_less_than():
+        actions.auto_insert(' < ')
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(' <= ')
+
+    def code_operator_and():
+        actions.auto_insert(' && ')
+
+    def code_operator_or():
+        actions.auto_insert(' || ')
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(' & ')
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(' | ')
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(' ^ ')
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(' << ')
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(' <<= ')
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(' >> ')
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(' >>= ')
+
+    def code_operator_object_accessor():
+        actions.auto_insert('.')
+
+    def code_block():
+        actions.auto_insert('{\n\n}')
```

Comment:
I believe the intent of `block` is to have the cursor within the braces when it ends.